### DISCUSSION TITLE
How to notify about a temporary road, path, street closure as a non-mapper?

### DISCUSSION THREAD

**Discussion by Filbuntu** [osmandapp/OsmAnd#13836] [Maps Related]
What is the best way to inform mappers or OsmAnd developers about a temporary road, path, street closure as a non-mapper?

From my experience notes about closed roads on OpenStreetMap often don't translate into real information that can be picked up by OSM navigation apps like OsmAnd or Magic Earth.

Thanks for any help!

**Comment by vshcherb** [comment]
Unfortunately it's only possible by a mapper to set access:conditional tag.

**Reply by Filbuntu** [reply]
Thank you @vshcherb for your response! 
Is there a way to inform mappers or OsmAnd developers about a temporary road, path, street closure as a non-mapper (apart from adding a note to OSM)?

**Reply by tommycrock** [reply]
A note is probably the best option. Alternatively you could see if there's an active forum or mailing list for your part of the world. Have a look under OSM community here https://forum.openstreetmap.org/

**Comment by Lee-Carre** [comment]
Mapping temporary information is discouraged (because some users don't update their map data often or at all). The only possible exception, from what I've read, is using `*:conditional` tagging.
This might explain part of why your notes don't receive much attention. Other parts may be that there's much else to do than figuring out how to tag temporary restrictions.

As a non-mapper, the best mechanism is notes. However, keep in mind that you're essentially asking / expecting others (who have their own priorities & resource limitations) to make the changes for you.

Much like Wikipedia; if you want to improve something, then learn how and spend the time to do so.

A loftier approach, would be to persuade whichever government department is responsible for such temporary closures, to make updating OSM part of their workflow.
However, a word of caution; they may not do a good job of it, because it'll likely be treated as yet another burden. My own government can't even keep its own data updated, so I wouldn't trust them with (making quality changes to) OSM.

Longer-term; when a city is fully mapped, then maybe more attention will be given to the likes of roadworks.

Bottom-line: if you want to see improvement, then help by contributing.
It needn't be taxing. An excellent way to start (besides dropping notes) is with StreetComplete. It asks simple questions (though, they're not always simple to answer; reality is messy & complex) in a pretty GUI, without requiring knowing the underlying clockwork.
StreetComplete is a good way for a newbie to start, and learn more, in the journey toward proficiency.
This may help relieve burden from other mappers, and thus be more likely to deal with nice-to-haves (like properly tagged temporary changes).

As for notes; ensure that you include as much specific info as possible. Often, when notes are vague, they're more likely to be ignored. StreetComplete can help; it enables attaching photos to notes; so you could take enough to convey all the details:
* is the way fully or partially closed?
* if partial, to what extent & how?
* starting & ending dates?
* are only some types of traffic restricted, or all?
* what's the nature of the work being done?
* where does the work zone start & end?
* relevant contact details (& name of the operator)?
* is there a URL to more info (such as a diagram)?
* anything else which may be relevant

**Reply by Lee-Carre** [reply]
Alternatively, you could pay a willing enthusiast to make the changes you desire.

**Reply by Filbuntu** [reply]
Dear @Lee-Carre 
Thanks a lot for your comprehensive and helpful answer.
Maybe I share with you the background of my query. Like you I am a "public-interest enthusiast & advocate" and open source enthusiast. I would love to see that Open Source Apps and services work as well or even better than the proprietary stuff. OsmAnd would be a great tool if it can be used by the average person. But if major changes or closures are not reflected in the map (quickly), OsmAnd can not really be used for navigation and therefore can not replace certain "big" map apps. I would love to promote OsmAnd and MagicEarth and others to my friends and family but if they get stuck because of road closures, they would just delete the app again and go back to the default installed app :(.
Local OsmAnd users (like me) know about the road closures they would notify OSM about and would find their way easily and they probably don't use it for their local trips. But if a "foreign" OsmAnd-user navigates through "this area", he would be happy if he would be notified about the road closure early on and can find a good alternative. 
**The lower the bar for people who don't know much about OSM to report any road closure, the quicker these are implemented in OSM and updated in OsmAnd the more people would be happy with OsmAnd.** This is my goal and I wondered whether there is already an easy way for non-mappers to notify the advanced guys to change something. And it would be helpful road closures could be prioritized to other more detailed changes.
**But as it seems not so easy to notify people and make these changes, I wonder whether this could be changed in order for OsmAnd being able to compete with the "big" Map apps.** (If this is a desired goal or even possible.)

**Reply by Filbuntu** [reply]
> Mapping temporary information is discouraged (because some users don't update their map data often or at all). The only possible exception, from what I've read, is using *:conditional tagging.

I know: [Support access:conditional, oneway:conditional, maxspeed:practical:conditional restrictions](https://github.com/osmandapp/Osmand/issues/1794)
[How to map path temporary closed](https://github.com/osmandapp/OsmAnd/discussions/12720)
https://wiki.openstreetmap.org/wiki/Good_practice#Don.27t_map_temporary_events_and_temporary_features
"it should at least be expected to remain unchanged in the next few weeks when you map it."

There are certain road closures for some months and this information should be added/mapped, I think.

**Reply by Lee-Carre** [reply]
Ah, your ideas mirror my own.

That wasn't apparent from your OP. It might have been wise to include such context. However, in that case …

Firstly, the distinction must be made between
* how to map temporary restrictions (tagging scheme, and such), particularly in a machine-readable form which routing algorithms (or other software) could ‘understand’
* a mechanism for normie users (non-mappers) to report a newly discovered restriction

(Which implies the necessary distinction between OSM & OsmAnd. Better to solve a problem for all of OSM than only one app.)

From my own research, having a comparable wish to yours (of OsmAnd auto-routing around roadworks 😃), the first part is mostly settled.

The trouble comes with the second part, due to the nature of OSM being very different to that of proprietary map houses.

For detail, read my (several) responses in #11409 (since many of the same factors (or underlying principles) apply to both that, and this; all that changes is the type of features / data).

At best, an in-app widget would just drop a note. Then (like above) it's down to actual mappers making the requisite change. Same problem (bottleneck), doesn't really solve anything.

Instead, you could take on responsibility of such tasks (evidently no-one else is, in your area). Otherwise, it's just a hope that someone else will do the work, instead.

A few things too be mindful of:
* quite a few apps only update once per month, so you'll have to be on the ball with applying changes promptly, in advance, to have any positive effect
* not all apps will use such data (Organic Maps strips a lot out of the raw data, for example)
* specific tagging for different types of traffic; while a road may be closed to vehicles (for repair), it's often still open to pedestrians
* I think there were a few more, which slip my mind; I'll add them in a future edit if they come back to me

Much more generally; while I'm all for improvement without undermining what OSM is (i.e. trying to fundamentally change it to be more like some proprietary mega-corp), beyond a point one must accept that a lot of users don't care about their freedom or anything beyond the practical. That's a problem of education. Even then, you won't persuade everyone.

I say this from experience; both as an advocate of tech-freedom in general, and from my conversations with various folks while out surveying.
I also say it as one for whom a minor motive for improving OSM (the major motives are all matters of principle) is for it to be *practically* better, and thus it's easier to persuade people to abandon Google (and similar). The catch is that this may be fickle if they don't grok the deeper reasons why they should value liberty. However, the proprietary players are likely to lose, simply because of the amount of work involved in maintaining an accurate global map.

Keep in mind, though, that while marking formal & planned road closures may be viable, if you also expect OsmAnd to match proprietary apps with the likes of
* traffic / congestion data
* delays due to a collision, police check, or similar
* temporary / mobile speed cameras (including a cop with a radar gun)
* …

then those will likely need a different approach than for geographical data.

For example, traffic data is often silently collected by proprietary apps from people using said app for navigation. So, regardless of technicalities, the likes of (opt-in) consent & privacy must be given due consideration. Funding (for hosting of infrastructure, for such en-masse real-time data collection from thousands of users) is then another concern; while proprietary map houses profit from their users, OSM not only receives no revenue from users (donations not withstanding, but they're optional), more users means more cost (think server capacity for serving all the requests in a timely fashion; from what I've read, OSM runs ‘lean’ on capacity, compared to what might be sensible for a comparable operation by a proprietary house; consider that `planet.osm` downloads are strongly encouraged to be  done via torrent, rather than HTTP from OSM servers).
Part of the implications re funding are that proprietary houses can afford to pay staff to make changes for users, based on mere reports, whereas I think all OSM folks (including members of Working Groups & governance (OSM Foundation)) are unpaid volunteers.

Plus, all this would have to be in a way that's long-term sustainable. That's no small challenge.

Software / datasets aren't magic. This goes for the above, and what's to follow.

If taking on the responsibility (for your area) is daunting
* well, that may be why others haven't, either
* someone's got to do it; since you're interested & informed, that might make you most qualified (congratulations)
* I'm the sole surveyor (and one of a tiny handful of mappers) in my area, so I haven't that much sympathy; vast amount to do around me (including basic fundamentals, like GNSS traces for aligning imagery, let alone actually mapping even major features (most buildings)); the solution is to just get on with mapping
* one step at a time; if mapping is a daunting prospect, then real-time data is a pipe-dream
* from your descriptions, if e.g. some road is always closed every winter, then that `*:conditional` need only be set once, unlike ephemeral restrictions such as roadworks; so, start with the permanent / recurring ones, before the ephemeral ones

---

Now that I've covered the general strokes, on to some specifics:

> The lower the bar for people who don't know much about OSM to report any road closure, the quicker these are implemented in OSM

Several things to unpack & address, here.

Firstly, the bar is already pretty low; OSM Notes. Don't even need an account to open a note; can be anonymous.

As I advocated in #11409, there is merit in some kind of entry exam. If users aren't interested enough to discover Notes, then contributing isn't for them.

Again, OSM doesn't have an army of droids to make changes for users based on reports.
Else, the equivalent in OSM is the large population of volunteers, and OSM Notes. But, volunteers have their own constraints & priorities, so there are no deadlines for solving notes; depends who's interested and has the time (hence pointing out that you can help).

This is an example of what I meant by changing fundamentals (for the worse); OSM being open and a community are great strengths. It's not some ‘product’ to get-rich-quick.
If that's inconvenient to some, then I'd argue that (contributing to or complaining about) OSM isn't for them (but they're free to use it on the understanding that they're not a customer, or entitled to anything). If they don't like _this_, then they're welcome to learn how to contribute, if they wanna see improvement.

Your statement (that I quoted) also assumes that mere reporting will lead to speedy resolution. Not at all.

Again, volunteers with their own priorities & interests.

There's plenty of notes which remain open for years. Else, some lack sufficient info for anyone but a surveyor to solve. Others are either irrelevant or bizarre; imagine a note on open water, which simply read “house”. That's probably just gonna be closed as spam / noise / erroneous.

Even with quality notes crafted carefully by the author; maybe solving it is a lot of work. So, it's often skipped for simpler ones.
Time required shouldn't be underestimated; I've done 12-hour surveying sessions (yes, all day, straight through, constantly), and tackled only ~40 problems, so felt like I didn't get much done.
Once, I spent multiple days on a single `power=minor_line` (& associated `power=pole`s).
Sometimes I've spent multiple hours at the same site.

Consider, also, that without some mechanism to prevent it (of which I have no idea how it would work), until the requested change is implemented & distributed, users will continue to drop duplicate notes about the same problem. I've seen this happen, with other features.

> This is my goal and I wondered whether there is already an easy way for non-mappers to notify the [experienced] guys to change something.

The easy way is via notes. But, opening a note doesn't imply anything else (such as even when it might be looked at).

The more experienced a mapper is, the more types of work he's capable of.
Besides, many mappers have particular interests (infrastructure, nautical-related, particular types of retailers, golf courses, sports, and whatever else).

I suggest changing your goal to become a mapper capable of making the changes you wish to see. Be that advanced, or whatever you wish to call it.

That you (and anyone else) is free & able to edit OSM is its primary strength.

Aside from benefitting others, it's also a stimulating discipline.

> it would be helpful [if] road closures could be prioritized [over] other more detailed changes.

Road closures **are** detailed changes.

Why that detail over anything else? There are areas which are very poorly mapped. I've already mentioned that in my own area, we lack GNSS traces, most building, and many minor roads (think of housing estates).

Everyone wants their favourite features prioritised. Who decides what is top priority?

This, again, is the beauty of a system open to all; you decide what you work on (just as others do).

As mentioned, many mappers work on particular types of features. Some may have no experience with detailed highway work.

Why should any individual mapper work on what you want, before his own interests (keep in mind that many work on mapping humanitarian-aid areas, such as those facing disasters, or war, in order to help locals & organisations there to helptgd locals; in such cases, people's lives are at stake, just to put things in perspective)?
I, myself, am prioritising defibrillators (for what should be obvious reasons), at the moment. Particularly since they are too small to see in imagery (and so require survey), there's no single dataset (let alone a map) of all of them (at least until I've finished them all). It's no good having a bunch of defibrillators dotted about, if people can't actually **find** them in an emergency.
After that, among other things (correcting mistagged elements / features), there's ~100 of each phone boxes and post boxes which need resurveying, since many have been closed (without a complete public list). This is important so that the map isn't misleading (showing something there, where there isn't) and possibly causing users to go in the wrong direction when seeking the nearest, if that one doesn't exist but the other one does.
Then, I'll be adding recycling collection sites (again, so that they're easy for people to find, because there's no map of them, currently).
Next might be EV charging stations.

There are businesses which have added a note requesting to be mapped, which are already several years old.

I also have a lot of `wheelchair[:*]` tagging, to do (I have access to non-public sources of info about which places are / aren't accessible). This being so that the disabled can easily find accessible places, and avoid inaccessible ones. OSM is optimal for distributing accessibility information.

Keep in mind that I'm also not overly interested in any of those categories. I'd rather be mapping the likes of infrastructure. But, I recognise that some things are more important, and choose to focus on them first.

Given time (again, as lone surveyor on my island), I'll get around to the nice-to-have features.

So, while roadworks would ad beneficial to map, and I agree with the outcome you describe (of apps having detailed info for routing, to offload that whole task funny drivers), and it's something I've researched how to map, they're simply not a priority compared to so much else.

It's not because I, or others, think it unworthy or pointless; but that there's plenty of more basic features to map, first. A road needs to exist (on the map), before it can be tagged as closed.

An example I focus on (until it actually opens) is a [new road in town](//www.openstreetmap.org/way/912950093), currently under construction. I'm trying to catch it very shortly after it's opened to the public, or ideally when it's being opened. So, I check every few days (it's already overdue it's completion date, so Real Soon Now™). Besides general accuracy & freshness (as for any feature), it's to serve as a pedestrian shortcut. Thus, will affect routing (especially for those on foot).

Another feature I give attention to, during surveying sessions, is footpaths (shortcuts for pedestrians). Often not obvious on imagery, and some are well-hidden (so have to go looking for them).
Else, marking stairs for the benefit of those with wheels.

So, roadworks? Maybe later.

Plus; let's imagine that (somehow, magically) everything is added, and even that it's recently updated / checked. How am I, alone, supposed to keep up with all the many various road closures going on (even on a small island)?
That could easily be a full-time occupation (and I would like at least a little time to do other things; especially when the weather makes surveying difficult / unpleasant / impractical).

So, then, it becomes; which closures do I tackle first? I can't do them all at once. Why one ever another (even if I put any weight on the demands of non-mapper users)?

Each mapper has only finite resources (including time).

If someone's got a good argument (not simply their personal preference or what directly benefits them) as to why I should change my priorities, then I'll consider it. Otherwise, I'll continue as I see fit.

> it seems not so easy to notify people and make these changes

Notifying is easy (OSM Notes), what's challenging is editing (accurately; sometimes I've found real head-scratchers: ‘how the hell do I map / tag this?!’).

But, to compare, think how long it takes to make a film, versus how long to watch it.

Having changes made; relatively easy;
1. open an account
2. learn how to do quality work
3. get cracking

Again, what seems to be expected is for others (unpaid volunteers) to do the desired work, promptly, as a priority.
Well, like I suggested:
* make the changes yourself
* pay someone to make them

In either of those cases, no need to notify anyone else.

This same principle / concept applies on Wikipedia; when merely asking others to make changes, they may never happen (or take a long time). Because editors are busy with their own changes.

There's always demand for improvement. To meet it (in a timely fashion) requires an increase in the count of proficient contributors.

One reason why I do surveying, is that its sorely needed. All other locals are armchair mappers.
However, surveying takes even more time (think of logistics, and practicalities like weather); so my list of targets is ever-growing, and it'll be a while before I get to any of the nice-to-have features.
Just trying to keep up with changes due to the likes of construction and retail attrition (one operator leaves, another moves in) is plenty challenging (for a handful of locals).

Besides helping the cause, contributing will give one by appreciation of quite how much is involved and some of the challenges.

> I wonder whether this could be changed in order for OsmAnd being able to compete with the "big" Map apps. (If this is a desired goal or even possible.)

While proprietary maps may compete with OSM, OSM doesn't compete with them. OSM isn't trying to achieve monopoly, or other profit-centric goals. It's primary goal isn't popularity, but freedom.
Corporations seek to become monopolies (and trap their users). OSM does not; quite the opposite.

To try to apply such narrow models & thinking to a public-interest project (since you claim to be an advocate of them) is to miss important fundamentals, and neglect the very nature of such projects.

Learn the early history of OSM; it started as a response to Ordinance Survey (which is tax-funded) forbidding the public from using the very data which they've already paid for. Being allowed to view a map doesn't count as (fully) using the data; correcting it, updating it, adding to it, repurposing it, sharing it, and so on.
OSM was an escape from that narrow model based on restriction.

With that out of the way; to address your ‘question’. I don't see how it could be changed without fundamentally altering the nature of OSM for the worse. That seems undesirable; undermining OSM's core virtues.
Again, if someone has a compelling argument, then let's have it.

The proprietary players aren't a concern; they'll eventually lose (as tends to happen when proprietary tries to compete against libre). Think about Google trying to get users to submit updates for them. Because Google has no hope of keeping up with a changing world.
I, for one, deliberately shun Google, on principle; because it wants the benefit of users helping it, without giving anything (e.g. it's dataset) back. I only contribute to OSM. I don't worry about the other players, either. They don't pay much attention to my island. I cited the new Moneypenny Lane, earlier; go compare what's shown on other maps in the same area; they're all outdated.
I've added a bunch of features which aren't on proprietary maps.
So, I don't give a damn (anyway) about proprietary data-hoarders, and don't have to.

Furthermore, OSM (& OsmAnd) is already widely used. It's certainly the ‘biggest’ in terms of freedom, and importance.
OSM is used by many sites & projects (some well-known), and I recall OsmAnd being popular as a navigation app.

OSM has something which proprietary players will never have; enduring respect for user-freedoms, and (as a result) a large & growing number of dedicated contributors.

Even if OSM weren't so wide-spread (like in the early days); well, just keep working on it.

The only reason why well-known names (Tom Tom, Facebook, Apple) are paying attention and starting to use OSM is because it's in their self-interest to (not because they have hearts of gold and have suddenly seen the light), and because much of the hard work has already been done (to produce a usable map). So, they're just jumping on the bandwagon. Not important. Ignore them; improve OSM for its own sake (as if it were a small hobby project).

Focus on preserving freedom, quality, the community collaboration, and the other attributes which made OSM virtuous.
Don't try to mutate it into some corporate monstrosity, just because it's now become popular.
In some places (speaking from experience), most folks still haven't heard of this 17-year-old project.

If you still prefer to think in terms of competition & combat, then read Sun Tzo's Art of War.
Much sage & relevant advice, which OSM is already doing.

**Comment by Filbuntu** [comment]
> A loftier approach, would be to persuade whichever government department is responsible for such temporary closures, to make updating OSM part of their workflow.

This would be a great idea! As far as I read in the newspaper the government department informed the big map app developer about the closures. Could there be a place for governments to easily inform OSM?

**Reply by Lee-Carre** [reply]
> the government department informed the big map app developer about the closures. Could there be a place for governments to easily inform OSM?

Well, there's a few differences.

With proprietary maps, all the transport / infrastructure department **can** do is inform / notify.

However, I would hope that they publish updated info, which anyone could check, regardless (for budding enthusiasts to use).
For OSM, publishing such info (somewhere that the public / mappers can discover it) is, in effect, the same as notifying.

Plus, for OSM, being a community, who or how, exactly, should they notify? With proprietary maps, that's much simpler.

However, with OSM, government agencies can do more than notify (which merely places the burden on volunteers). My point was that they could make it part of the procedures to have an employee sit down with editing software, and edit the map data themselves (again, like Wikipedia).
Though, again, that might not be the best idea for quality (I speak from experience 🤨🤬).

A first step, though, would be to ensure that your public servants are at least publishing relevant data (& in a timely fashion).
If it would help (as examples), I can cite the relevant info (not just a list, but also a map) published by my local bureaucrats.

That way, you or other mappers can take on the task of applying it to OSM data.

The only caveat would be copyright. Unless you're gonna survey each site, you'd have to copy from the published data. So, the license applied to the roadworks (or whatever) data must be compatible with OSM, or an explicit (written) waiver / permission must be granted.
Otherwise, it can only be used as reference (to find the sites), which must be surveyed.

Unless you're somewhere tiny, or very backward (which sounds like it isn't the case, for you), then your government likely does have that data in GIS format(s) already, even if they don't yet publish it. They'll need it to help keep themselves organised.

Ultimately; be the change that you want to see in the (world\|map).

**Comment by Filbuntu** [comment]
> Bottom-line: if you want to see improvement, then help by contributing.

I will do that :).


>  StreetComplete is a good way for a newbie to start, and learn more, in the journey toward proficiency.

I will look into this. Some links for others who would like to join:
https://wiki.openstreetmap.org/wiki/StreetComplete
https://f-droid.org/en/packages/de.westnordost.streetcomplete/
https://github.com/streetcomplete/StreetComplete

**Comment by Filbuntu** [comment]
> Alternatively, you could pay a willing enthusiast to make the changes you desire.

How would someone easily find an enthusiast? 

As written above, most people would probably not need the temporary map change for themselves but would like to contribute to the community but don't know how to change OSM by themselves. If there is a way to do this without much hassle and if these notifications/notes could be prioritised (e.g. relevant for navigation), this would be great.

**Reply by Lee-Carre** [reply]
> How would someone easily find an enthusiast?

There's many of them on OSM.

Perhaps post in the forum. Else, ask on the help site how best to go about advertising your offer.

Don't forget to donate to the OSM project itself; hosting costs money.

> most people would probably not need the temporary map change for themselves but would like to contribute to the community but don't know how to change OSM by themselves.

Well, there's a whole wiki of documentation.
The default editor (iD) has a basic tutorial.

Start with the basics. Expend from there. Many mappers (myself included) learned this way.

> If there is a way to do this without much hassle

Learning anything (let alone mastering it; which takes ~10,000 hours), and doing quality work, always takes dedication & time. There is no shortcut. Those who claim otherwise are either charlatans or trying to sell you something.

> if these notifications/notes could be prioritised (e.g. relevant for navigation), this would be great.

See above; why would / should they be (above anything else)?

If you want them implemented, then contribute.

Making demands of volunteers probably won't lead anywhere. Everyone's got shit to do.

**Comment by Filbuntu** [comment]
Hi @Lee-Carre 
Just an update three years later! I have followed your advice and have been contributing to OSM regularly! 
Thank you for sharing your thoughts in 2022.
Have a great day
